### ISSUE TITLE
Lower barrier of entry for submission

### ISSUE THREAD

**Issue by franky47** [peterpeterparker/discoverweekly.dev#36]
At the moment, submissions are only possible via a fork/PR/merge process, which is cumbersome.

Some ideas to automate submissions:
- GitHub issue, monitor comments with an Action
- Web UI

**Comment by peterpeterparker** [comment]
I most probably not gonna build a Web UI soon (too many side projects 😉). Of course not against a PR if anyone would jump in.

To the contrary, the "GitHub issue, monitor comments with an Action" sounds like an interesting option and learning.

Thanks for the idea and issue!

**Comment by franky47** [comment]
Yes, the web UI is probably the option that requires the most work, I'm not a fan either (plus having to deal with captcha, spam etc).

At the end of the day, having static Markdown files for submissions is good. The way I see an integration with comments:
- Have a pinned issue, for example named "Submissions"
- Comments to that issue trigger a GitHub Actions run that:
  - Verifies the formatting and aborts if invalid
  - Pulls information about the user from the GitHub API (can be edited by the user afterwards with a PR)
  - Generates and commits the Markdown file (see _"Validation"_)

Cases not covered:
- Editing the comment (edits the file)
- Editing the file (doesn't edit the comment)

## Validation

Just like you approve submissions via PRs, you'd need a way to approve those as well. I see two ways to do that:
- Liking the comment (👍) triggers the workflow _(approve beforehand)_
- The workflow runs for all comments, but open a PR rather than landing on `main` _(approve afterwards)_

**Comment by peterpeterparker** [comment]
Sounds super cool! I am not  the best when it comes to such automation, it's a bit challenging and therefore interesting 😉.
Being said, here too open to contributions.

**Comment by franky47** [comment]
I've got a basic proof of concept in [`franky47/actions-laboratory`](https://github.com/franky47/actions-laboratory):
1. The [workflow](https://github.com/franky47/actions-laboratory/blob/main/.github/workflows/commentToPR.yml) listens for issue comments
2. An [example comment](https://github.com/franky47/actions-laboratory/issues/1#issuecomment-796585769) was posted
3. The [Actions run](https://github.com/franky47/actions-laboratory/runs/2084829515?check_suite_focus=true) generated a [Pull Request](https://github.com/franky47/actions-laboratory/pull/2)
4. Which created [a file with the content of the comment](https://github.com/franky47/actions-laboratory/blob/main/commentToPR/796585769.md)

Todo:
- Lock workflow to a specific issue number
- Add validation to avoid spam
- Test comment edition with PR update

As a side-note, I'm definitely going to write about how this hack can make blog posts easier. 😀

**Comment by peterpeterparker** [comment]
Wow that's such a cool hack 🔥

**Comment by franky47** [comment]
This is seriously cool. It can also show the action progress/status using reactions on the original comment:

![reactions](https://user-images.githubusercontent.com/1174092/110794084-a5d64680-8275-11eb-9d88-fba1cc8972db.gif)

**Comment by peterpeterparker** [comment]
Hahaha c'que c'est bien!
